curl/tests/response_test.cc

```diff
@@ -0,0 +1,63 @@
+#include <catch.hpp>
+#include "mock_curl.hpp"
+#include <leatherman/curl/client.hpp>
+#include <leatherman/curl/request.hpp>
+
+using namespace std;
+
+namespace leatherman { namespace curl {
+
+    struct mock_client : client {
+        curl_handle const& get_handle() { return client::get_handle(); }
+    };
+
+    TEST_CASE("curl::response") {
+        SECTION("Headers should be addable and retrievable from the response") {
+            response test_response;
```

Comment:
You could still pull `response test_response;` out of every SECTION into the TEST_CASE. CATCH ensures a new object is created for each SECTION.